src/ooxml/java/org/apache/poi/POIXMLDocumentPart.java

```diff
@@ -368,6 +368,15 @@ public final POIXMLDocumentPart createRelationship(POIXMLRelation descriptor, PO
      */
     protected final POIXMLDocumentPart createRelationship(POIXMLRelation descriptor, POIXMLFactory factory, int idx, boolean noRelation){
         try {
```

Comment:
Update the javadoc and/or add a comment if `idx<0` has a special meaning. It might be worth moving this "find the next highest available reference number" code into its own private function.